[FrameworkBundle][Translations] Fix translation commands

kevin-verschaeve opened the pull request:
Hi,

I'm not sure about the target branch, but this bug seems to be present since we have the new `translations` directory in the project root. 

This add the search for a `translations` directory in the project root, **and** enable to have translations with `.yaml` extension, for now only `.yml` is working.

| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Please tell me how can I test this.

baptistedonaux commented on src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php:
```diff
@@ -113,7 +113,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
-        $transPaths = array($kernel->getRootDir().'/Resources/');
+        $transPaths = array(
+            $kernel->getRootDir().'/Resources/',
+            $kernel->getProjectDir().'/',
```
There is one problem here. These paths are used to generate `views` and `translations` folders paths (see L275 & L295). With Flex, `views` is renamed by `templates`, so translations could be extracted but translation keys used in templates won't be extracted.

https://github.com/symfony/symfony/issues/24807#issuecomment-341669603

baptistedonaux replied:
I propose you a simple solution in PR.

chalasr commented:
Translation paths are already [collected by framework extension](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L934). Couldn't we inject them in the command instead of discovering them twice?

baptistedonaux commented:
@chalasr I think is a better solution but the current implementation has probably implemented for specific reason.

chalasr commented:
@baptistedonaux At time it has been introduced (#10368), a `bundle` argument was required by the command. 
In https://github.com/symfony/symfony/pull/13443, `%kernel.root_dir%/Resources/translations` was added as fallback directory (), still letting only 2 possible dirs, both available via existing parameters. So it was just not worth it.

Later, a `paths` option was added to the `framework.translator` config node (#14561). 
Currently these `paths` are not taken into account by translation commands.
I think we should fill this gap on 3.4 where these commands are registered as services and have constructors. I will take care of the PR.

chalasr reviewed (approved):
👍

chalasr commented on src/Symfony/Bundle/FrameworkBundle/Resources/config/translation.xml:
```diff
@@ -33,7 +33,7 @@
 
         <service id="translation.loader.yml" class="Symfony\Component\Translation\Loader\YamlFileLoader" public="true">
-            <tag name="translation.loader" alias="yml" />
+            <tag name="translation.loader" alias="yaml" legacy-alias="yml" />
```
hmm, is that another bugfix?

kevin-verschaeve replied:
yes. It allow to find `.yml` and `.yaml` files for translations. I let this in this PR because I think it is part of the command fix

kevin-verschaeve commented:
Hi, thanks for your review.

So what needs to be done ? Should I update the code to get the paths from the `path` options of the translator ?

baptistedonaux commented:
Probably the best way. @chalasr ?

chalasr commented:
From me this is OK regarding commands, it just looks into `%kernel.project_dir%/translations` as of 3.4, that seems to make sense. I will have a look to custom paths handling, it has to be done on another branch.

kevin-verschaeve committed:
[FrameworkBundle][Translations] Fix translation commands

Search for translations also in the new translations dir at the project root
Search for .yaml file as well as .yml

chalasr reviewed (changes requested):
For now the `%kernel.project_dir%/translations/` dir has been abandoned in favor of (https://github.com/symfony/recipes/pull/249) so this should be updated accordingly, but that's still in discussion in https://github.com/symfony/recipes/pull/253. Let's wait

kevin-verschaeve commented:
I should close this PR in favor of https://github.com/symfony/symfony/pull/25065, seems to be more updated and better implemented

nicolas-grekas commented:
Looks like we kinda missed your PR... this would mean #25065 should be rebased on 3.3? @yceruto WDYT?

yceruto commented:
Hmm, not sure how right now ... I know 4.0 structure + flex is compatible since 3.3, but the solucion here is a _little_ tricky. IMO 3.4 is a better place to migrate to the new structure as there we have the default paths and the new overriding conventions related to `translations/` and `templates/` directories.

I suggest to keep structure as is in 3.3 and then change it in 3.4 as a progressive migration path in our projects. Wdyt?

chalasr commented:
Supporting the new directory structure starting from 3.4 makes sense to me.

nicolas-grekas commented:
So, shall we close this PR?

chalasr commented:
Yea, closing in favor of #25065 which is more advanced. 
Thank you for starting the work @kevin-verschaeve.

kevin-verschaeve commented:
Fine for me too.

What about https://github.com/symfony/symfony/pull/24590/files#diff-ace3ed7f9c8dd7ded3aa562e3c25fa92R36 
translation loader should read `.yml` and `.yaml` files for translations. It is not the case here
